src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ServiceCategoryService.kt

```diff
@@ -11,4 +13,25 @@ import java.util.UUID
 @Transactional
 class ServiceCategoryService(val repository: ServiceCategoryRepository) {
   fun getServiceCategoryByID(id: UUID): ServiceCategory? = repository.findByIdOrNull(id)
+  fun getServiceCategoryByIDAndContractReference(id: UUID, contractReference: String): ServiceCategory? {
+    val serviceCategoryWithAllDesiredOutcomes = repository.findByIdOrNull(id)
+    val serviceCategoryWithFilteredOutcomes = serviceCategoryWithAllDesiredOutcomes?.copy()
+    serviceCategoryWithFilteredOutcomes?.desiredOutcomes = serviceCategoryWithFilteredOutcomes?.desiredOutcomes?.filter { isDesiredOutcomeValidForContract(it, contractReference) }.orEmpty()
+
+    return serviceCategoryWithFilteredOutcomes
```

Comment:
```suggestion
 return serviceCategoryWithAllDesiredOutcomes?.copy(
      desiredOutcomes = serviceCategoryWithAllDesiredOutcomes.desiredOutcomes.filter {
        isDesiredOutcomeValidForContract(it, contractReference)
      })
```
I'm not really sure if this is neater and worth testing. But thought it was interesting that you can actually pass variables into the `.copy()` method to overwrite. So up to you whether you want to use or not